src/test/libcephfs/perfcounters.cc

```diff
@@ -0,0 +1,54 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#include "include/compat.h"
+#include "gtest/gtest.h"
+#include "include/cephfs/libcephfs.h"
+#include "common/ceph_json.h"
+#include "include/utime.h"
+
+#include <string>
+
+using namespace std;
+
+TEST(LibCephFS, ValidatePerfCounters) {
+  struct ceph_mount_info *cmount;
+  ASSERT_EQ(0, ceph_create(&cmount, NULL));
+  ASSERT_EQ(0, ceph_conf_read_file(cmount, NULL));
+  ASSERT_EQ(0, ceph_conf_parse_env(cmount, NULL));
+  ASSERT_EQ(0, ceph_mount(cmount, "/"));
+
+  char *perf_dump;
+  int len = ceph_get_perf_counters(cmount, &perf_dump);
```

Comment:
Does this API fail on an uninitialized or unmounted client? We don't have a check for that in the libcephfs/Client API.

Reply:
Well, I think this API would still work when the client is not initialized or unmounted. Do we want to add that guard?

Reply:
I wasn't actually sure what would happen if you don't `::init` the client before calling this. It's part of why I'm asking.

Reply:
OK, I see this would require the client to be mounted. Commenting that part out from the test crashes the client (invalid `cct`).

```
#0  Client::get_perf_counters (this=0x0, outbl=outbl@entry=0x7fffffffb5e0) at /home/vshankar/ceph/src/client/Client.cc:17495
17495     return cct->get_admin_socket()->execute_command(cmd, inbl, err, outbl);
(gdb) p cct
Cannot access memory at address 0x8
```
I think this API should be allowed for initialized/mounted client (the perf counters are registered in Client::init() anyway).